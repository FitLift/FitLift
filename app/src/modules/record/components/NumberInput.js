import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { TextInput } from 'react-native';

export default class NumberInput extends PureComponent {
  static propTypes = {
    value: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]).isRequired,
  };

  render() {
    return (
      <TextInput
        maxLength={2}
        keyboardType="numeric"
        value={`${this.props.value}`}
        style={{
          borderColor: 'gray',
          borderRadius: 4,
          borderWidth: 1,
          fontSize: 15,
          height: 30,
          textAlign: 'center',
          width: 30,
        }}
      />
    );
  }
}
