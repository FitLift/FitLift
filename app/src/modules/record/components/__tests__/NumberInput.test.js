import React from 'react';
import renderer from 'react-test-renderer';
import NumberInput from '../NumberInput';

describe('render tests', () => {
  it('doesnt crash', () => {
    const rendered = renderer.create(
      <NumberInput
        value={5}
      />,
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
