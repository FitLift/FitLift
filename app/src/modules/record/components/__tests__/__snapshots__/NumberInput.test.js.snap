// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render tests doesnt crash 1`] = `
<TextInput
  allowFontScaling={true}
  keyboardType="numeric"
  maxLength={2}
  style={
    Object {
      "borderColor": "gray",
      "borderRadius": 4,
      "borderWidth": 1,
      "fontSize": 15,
      "height": 30,
      "textAlign": "center",
      "width": 30,
    }
  }
  value="5"
/>
`;
